src/ripple/app/ledger/Ledger.cpp

```diff
@@ -372,13 +372,15 @@ getTransaction (Ledger const& ledger,
         txn = Transaction::sharedTransaction (item->peekData (), Validate::YES);
     else if (type == SHAMapTreeNode::tnTRANSACTION_MD)
     {
-        Blob txnData;
-        int txnLength;
-
-        if (!item->peekSerializer().getVL (txnData, 0, txnLength))
-            return Transaction::pointer ();
-
-        txn = Transaction::sharedTransaction (txnData, Validate::NO);
+        try
+        {
+            auto sit = SerialIter{item->data(), item->size()};
+            txn = Transaction::sharedTransaction(sit.getVL(), Validate::NO);
+        }
+        catch (...)
+        {
+            return {};
```

Comment:
My recommendation:  Just remove the try/catch here and let exceptions propagate out.